src/input/element-input.js

```diff
@@ -46,6 +51,40 @@ function scalarTriple(p1, p2, p3) {
     return _sct.cross(p1, p2).dot(p3);
 }
 
+// calculates the distance from a point to a 3D rectangle defined by its corners
```

Comment:
would it be more precise to use the ray, intersect it with the rectangle, get exact intersection point and return the distance to that point, instead of using corners?
I'm not entirely sure what this function does though .. what is 'distance from a point to a rectangle defined by its corners'? Is it a perpendicular distance or something?

Reply:
Oh! I believe I completely misunderstood the meaning of `element.worldCorners`. I thought it was the bounding box, so the code checks the closest point within the bounding box to another point. The algorithm was taken from the book Real-Time Collision Detection (same as other algorithms in this file), Chapter 5.

Thanks for the callout! I'll be able to simply this a lot and re-use the current implementation of intersectLineQuad.